dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/DefaultEventQueryPlanner.java

```diff
@@ -198,10 +206,14 @@ private List<EventQueryParams> groupByQueryItems( EventQueryParams params )
         {
             for ( QueryItem item : params.getItemsAndItemFilters() )
             {
+                AnalyticsAggregationType aggregationType = ObjectUtils.firstNonNull(
+                    params.getAggregationType(), fromAggregationType( item.getAggregationType() ) );
+
                 EventQueryParams.Builder query = new EventQueryParams.Builder( params )
                     .removeItems()
                     .removeItemProgramIndicators()
-                    .withValue( item.getItem() );
+                    .withValue( item.getItem() )
+                    .withAggregationType( aggregationType );
```

Comment:
The aggregation type had to be set on the query to become available for the "period grouper" method used for "last value" aggregation which comes later.